client/pdo/client/controller/commands/send.py

```diff
@@ -81,17 +81,26 @@ def send_to_contract(state, save_file, message, eservice_url=None, quiet=False,
     ledger_config = state.get(['Sawtooth'])
 
     if update_response.state_changed and commit :
-        try :
-            logger.debug("send update to the ledger")
-            extraparams = {}
-            if wait :
-                extraparams['wait'] = 30
-            txnid = update_response.submit_update_transaction(ledger_config, **extraparams)
-        except Exception as e :
-            raise Exception('failed to save the new state to the ledger; {0}'.format(str(e)))
+
+        contract.set_state(update_response.raw_state)
+
+        # asynchronously submit the commit task: (a commit task replicates change-set and submits the corresponding transaction)
+        try:
+            update_response.commit_asynchronously(ledger_config, wait=30, use_ledger=True)
+        except Exception as e:
+            raise Exception('failed to submit commit: %s', str(e))
+
+        # wait for the commit to finish.
+        # TDB: 1. make wait_for_commit a separate shell command. 2. Add a provision to specify commit dependencies as input to send command.
```

Comment:
TDB? TBD?? (If so TODO seems more common tag?)

Reply:
i always thought "To Be Done" as a common tag. interesting to know that it is not so common!

Reply:
create an issue rather than leave it in the code

Reply:
yes , issue 216